partiql-ast/src/main/java/org/partiql/ast/v1/AstRewriter.kt

```diff
@@ -0,0 +1,758 @@
+package org.partiql.ast.v1
+
+import org.partiql.ast.v1.Ast.explain
+import org.partiql.ast.v1.Ast.exprQuerySet
+import org.partiql.ast.v1.Ast.identifier
+import org.partiql.ast.v1.Ast.query
+import org.partiql.ast.v1.expr.Expr
+import org.partiql.ast.v1.expr.ExprAnd
+import org.partiql.ast.v1.expr.ExprArray
+import org.partiql.ast.v1.expr.ExprBag
+import org.partiql.ast.v1.expr.ExprBetween
+import org.partiql.ast.v1.expr.ExprCall
+import org.partiql.ast.v1.expr.ExprCase
+import org.partiql.ast.v1.expr.ExprCast
+import org.partiql.ast.v1.expr.ExprCoalesce
+import org.partiql.ast.v1.expr.ExprExtract
+import org.partiql.ast.v1.expr.ExprInCollection
+import org.partiql.ast.v1.expr.ExprIsType
+import org.partiql.ast.v1.expr.ExprLike
+import org.partiql.ast.v1.expr.ExprLit
+import org.partiql.ast.v1.expr.ExprMatch
+import org.partiql.ast.v1.expr.ExprNot
+import org.partiql.ast.v1.expr.ExprNullIf
+import org.partiql.ast.v1.expr.ExprOperator
+import org.partiql.ast.v1.expr.ExprOr
+import org.partiql.ast.v1.expr.ExprOverlay
+import org.partiql.ast.v1.expr.ExprParameter
+import org.partiql.ast.v1.expr.ExprPath
+import org.partiql.ast.v1.expr.ExprPosition
+import org.partiql.ast.v1.expr.ExprQuerySet
+import org.partiql.ast.v1.expr.ExprSessionAttribute
+import org.partiql.ast.v1.expr.ExprStruct
+import org.partiql.ast.v1.expr.ExprSubstring
+import org.partiql.ast.v1.expr.ExprTrim
+import org.partiql.ast.v1.expr.ExprValues
+import org.partiql.ast.v1.expr.ExprVarRef
+import org.partiql.ast.v1.expr.ExprVariant
+import org.partiql.ast.v1.expr.ExprWindow
+import org.partiql.ast.v1.expr.PathStep
+import org.partiql.ast.v1.graph.GraphMatch
+import org.partiql.ast.v1.graph.GraphPattern
+import org.partiql.ast.v1.graph.GraphQuantifier
+import org.partiql.ast.v1.graph.GraphSelector
+import org.partiql.value.PartiQLValueExperimental
+
+public abstract class AstRewriter<C> : AstVisitor<AstNode, C> {
```

Comment:
(self-review) `AstRewriter` written in Kotlin since it was slightly easier to create based on the sprout-generated `AstRewriter`. I believe it should have similar public API as if it was in Java but if not, I can migrate it to Java.

Reply:
I can check the decompiled bytecode, but this resides in the `java` source set btw.

Reply:
> but this resides in the java source set btw.

Yeah, I'll fix this in a later PR (probably the PR that deletes the sprout AST code). There's some other Kotlin sources in the `java` directory that I'll also move in that PR.